kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/Utilities.java

```diff
@@ -0,0 +1,254 @@
+
+    try {
+      Object result = configClass.getDeclaredConstructor().newInstance();
+      BoundMethod addResourceMethod =
+          DynMethods.builder("addResource").impl(configClass, URL.class).build(result);
+      BoundMethod setMethod =
+          DynMethods.builder("set").impl(configClass, String.class, String.class).build(result);
+
+      //  load any config files in the specified config directory
+      String hadoopConfDir = config.hadoopConfDir();
+      if (hadoopConfDir != null) {
+        HADOOP_CONF_FILES.forEach(
+            confFile -> {
+              Path path = Paths.get(hadoopConfDir, confFile);
+              if (Files.exists(path)) {
+                try {
+                  addResourceMethod.invoke(path.toUri().toURL());
+                } catch (IOException e) {
+                  LOG.warn("Error adding Hadoop resource {}, resource was not added", path, e);
+                }
+              }
+            });
+      }
+
+      // set any Hadoop properties specified in the sink config
+      config.hadoopProps().forEach(setMethod::invoke);
+
+      LOG.info("Hadoop config initialized: {}", configClass.getName());
+      return result;
+    } catch (InstantiationException
+        | IllegalAccessException
+        | NoSuchMethodException
+        | InvocationTargetException e) {
+      LOG.warn(
+          "Hadoop found on classpath but could not create config, proceeding without config", e);
+    }
+    return null;
+  }
+
+  public static Object extractFromRecordValue(Object recordValue, String fieldName) {
+    String[] fields = fieldName.split("\\.");
+    if (recordValue instanceof Struct) {
+      return getValueFromStruct((Struct) recordValue, fields, 0);
+    } else if (recordValue instanceof Map) {
+      return getValueFromMap((Map<?, ?>) recordValue, fields, 0);
+    } else {
+      throw new UnsupportedOperationException(
+          "Cannot extract value from type: " + recordValue.getClass().getName());
+    }
+  }
+
+  private static Object getValueFromStruct(Struct struct, String[] fields, int idx) {
+    Preconditions.checkArgument(idx < fields.length, "Invalid field index");
+    Object value = struct.get(fields[idx]);
+    if (value == null || idx == fields.length - 1) {
+      return value;
+    }
+
+    Preconditions.checkState(value instanceof Struct, "Expected a struct type");
+    return getValueFromStruct((Struct) value, fields, idx + 1);
+  }
+
+  private static Object getValueFromMap(Map<?, ?> map, String[] fields, int idx) {
+    Preconditions.checkArgument(idx < fields.length, "Invalid field index");
+    Object value = map.get(fields[idx]);
+    if (value == null || idx == fields.length - 1) {
+      return value;
+    }
+
+    Preconditions.checkState(value instanceof Map, "Expected a map type");
+    return getValueFromMap((Map<?, ?>) value, fields, idx + 1);
+  }
+
+  public static TaskWriter<Record> createTableWriter(
+      Table table, String tableName, IcebergSinkConfig config) {
```

Comment:
Do we expect `tableName` to be something other than `table.name()` or `table.toString()`?

Reply:
It could be different. The `tableName` parameter is used to look up table-specific configuration parameters and is always namespace + name. `table.name()` can have the catalog name prepended in some cases.